Installation instructions in documentation (#376)
skypyproject/skypy: docs/install.rst

############
Installation
############

This page outlines how to install one of the officially distributed SkyPy

> releases and it's dependencies, or install and test the latest development

Reviewer: ```suggestion
releases and its dependencies, or install and test the latest development
```

version.

From PyPI
---------

All  SkyPy releases are distributed through the Python Package Index (PyPI_).
To install the latest version use pip_:

.. code:: console

    $ pip install skypy

From conda-forge
----------------

All SkyPy releases are also distributed for conda_ through the `conda-forge`_
channel. To install the latest version for your active conda environment:

.. code:: console

    $ conda install -c conda-forge skypy

From GitHub
-----------

The latest development version of SkyPy can be found on the master branch of
the `skypyproject/skypy`_ GitHub repository. This and any other branch or tag

> can be installed directly from GitHub using pip:

Reviewer: ```suggestion
can be installed directly from GitHub using a recent version ofpip:
```

.. code:: console

>     $ pip install git+https://github.com/skypyproject/skypy@master

Reviewer: ```suggestion
    $ pip install skypy@git+https://github.com/skypyproject/skypy.git@master
```

Dependencies
------------

SkyPy is compatble with Python versions 3.6 or later on Ubuntu, macOS and
Windows operating systems. It has the following core dependencies:

- `astropy <https://www.astropy.org/>`__
- `networkx <https://networkx.github.io/>`_
- `numpy <https://numpy.org/>`_
- `pyyaml <https://pyyaml.org/>`_
- `scipy <https://www.scipy.org/>`_

Installing using pip or conda will automatically install or update these core
dependencies if necessary. SkyPy also has a number of optional dependencies
that enable additional features:

- `camb <https://camb.info/>`_
- `classy <https://lesgourg.github.io/class_public/class.html>`_
- `healpy <https://healpy.readthedocs.io/>`_
- `speclite <https://speclite.readthedocs.io/>`_
- `specutils <https://specutils.readthedocs.io/>`_

> 

Reviewer: Do we want to mention that all of these can be installed with `pip install skypy[all]`?

Testing
-------

Once installed, you should be able to import the `skypy` module in python:

.. code:: python

    >>> import skypy

You should also be able to check the installed version number using the `skypy`
command line script:

.. code:: console

    $ skypy --version

You may also want to run the unit tests, for example if you have installed the
development version or you use an unsupported operating system. The tests have
the following additional dependencies:

- `pytest-astropy <https://github.com/astropy/pytest-astropy>`_
- `pytest-rerunfailures <https://github.com/pytest-dev/pytest-rerunfailures>`_

> and can be run using pytest_:

Reviewer: All test dependencies can be installed with `pip install skypy[test]`.
